03_letters/luliangLi/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+type Pair struct {
+	key   rune
+	value int
+}
+
+func (p Pair) String() string {
+	return fmt.Sprintf("%s:%d", string(p.key), p.value)
+}
+
+func letters(s string) map[rune]int {
+	rlt := make(map[rune]int)
+
+	for _, char := range s {
+		val := rlt[char]
+		rlt[char] = val + 1
+	}
+
+	return rlt
+}
+
+func sortLetters(m map[rune]int) []string {
+	pairs := make([]Pair, len(m))
+	rlt := make([]string, len(m))
+
+	i := 0
+	for k, v := range m {
+		pairs[i] = Pair{k, v}
+		i++
+	}
+
+	sort.Slice(pairs, func(i, j int) bool {
+		return pairs[i].value > pairs[j].value
+	})
+
+	j := 0
+	for _, pair := range pairs {
```

Comment:
This loop shouldn't be required when you remove the `Pair` struct, but with this, you should be able to use the `index` value from the `range` result, rather than initialising a new `j` variable.
```go
for i, pair := range pairs {
  rlt[i], pair.String()
}
```